crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -634,35 +796,18 @@ impl Store {
 
     ///  Get the Identity of `user_id`
     pub(crate) async fn get_identity(&self, user_id: &UserId) -> Result<Option<UserIdentities>> {
-        // let own_identity =
-        // self.inner.get_user_identity(self.user_id()).await?.and_then(|i| i.own());
-        Ok(if let Some(identity) = self.inner.store.get_user_identity(user_id).await? {
-            Some(match identity {
-                ReadOnlyUserIdentities::Own(i) => OwnUserIdentity {
-                    inner: i,
-                    verification_machine: self.inner.verification_machine.clone(),
-                }
-                .into(),
-                ReadOnlyUserIdentities::Other(i) => {
-                    let own_identity =
-                        self.inner.store.get_user_identity(self.user_id()).await?.and_then(|i| {
-                            if let ReadOnlyUserIdentities::Own(i) = i {
-                                Some(i)
-                            } else {
-                                None
-                            }
-                        });
-                    UserIdentity {
-                        inner: i,
-                        verification_machine: self.inner.verification_machine.clone(),
-                        own_identity,
-                    }
-                    .into()
+        let own_identity =
+            self.inner.store.get_user_identity(self.user_id()).await?.and_then(|i| {
+                if let ReadOnlyUserIdentities::Own(i) = i {
+                    Some(i)
+                } else {
+                    None
```

Comment:
I think we have this pattern in a lot of places. Do you think pulling in or copy-pasting [`as_variant!`](https://docs.rs/as_variant/latest/as_variant/macro.as_variant.html) is worth it to make this shorter? I wrote that last year and never used it 😄

Reply:
Might be sensible, seems to be a simple enough macro.